glean-core/python/glean/_process_dispatcher.py

```diff
@@ -10,52 +10,46 @@
 This is used only by the PingUploadWorker (and the test-only feature to delete
 temporary data directories which has a potential race condition with the
 PingUploadWorker).
+
+This uses the lower level `subprocess` library rather than `multiprocessing` to
+avoid the unnecessary complexity and not place burdens on Glean's users to
+architect their application startup to be `multiprocessing` compatible on
+Windows.
 """
 
+import base64
+import pickle
+import subprocess
 import sys
-from typing import Optional, TYPE_CHECKING, Union
-
+from typing import Optional, Union
 
-if TYPE_CHECKING:
-    import multiprocessing  # noqa
 
-
-def _work_wrapper(func, args):
-    """
-    A wrapper to call a function in another process and convert its boolean
-    success value into a process exitcode.
-    """
-    success = func(*args)
-
-    if success:
-        sys.exit(0)
-    else:
-        sys.exit(1)
+from .subprocess import _process_dispatcher_helper
 
 
 class _SyncWorkWrapper:
     """
     A wrapper to synchronously call a function in the current process, but make
-    it have the same (limited) interface as if it were a
-    multiprocessing.Process:
+    it have the same (limited) interface as if it were a `subprocess.Popen`
+    object:
 
         >>> p = ProcessDispatcher.dispatch(myfunc, ())
-        >>> p.join()
-        >>> p.exitcode
+        >>> p.wait()
+        >>> p.returncode
         0
     """
 
     def __init__(self, func, args):
         self._result = func(*args)
-        self._joined = False
+        self._waited = False
 
-    def join(self):
-        self._joined = True
+    def wait(self):
+        self._waited = True
 
     @property
-    def exitcode(self):
+    def returncode(self):
         if not self._joined:
```

Comment:
```suggestion
        if not self._waited:
```

Reply:
This also highlights a lack of test coverage here, which I'll also address.